// @contentful/f36-utils/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('react');
var reactDom = require('react-dom');

function p({children:o,container:n}){let e=react.useRef(n),t=react.useRef(document.createElement("div"));return react.useEffect(()=>{e.current||(e.current=document.body);let r=t.current;return r.setAttribute("data-cf-ui-portal",""),e.current.appendChild(r),()=>{e.current&&e.current.removeChild(r);}},[]),t.current?reactDom.createPortal(o,t.current):null}var m=o=>{let{ref:n,keys:e,event:t="keydown"}=o,r=document,c=react.useCallback(a=>{Object.prototype.hasOwnProperty.call(e,a.key)&&e[a.key](a);},[e]);react.useEffect(()=>(n&&n.current&&(r=n.current),r.addEventListener(t,c),()=>{r.removeEventListener(t,c);}),[t,c]);};function d(o,n){let e={before:"",match:"",after:""},t=n.replace(/[.*+?^${}()|[\]\\]/g,"\\$&"),r=new RegExp(`(?<before>.*?)(?<match>${t})(?<after>.*)`,"i"),c=o.match(r);return c&&(e.before=c.groups.before,e.match=c.groups.match,e.after=c.groups.after),e}function b(o,n=1){let e=parseInt(o.slice(1,3),16),t=parseInt(o.slice(3,5),16),r=parseInt(o.slice(5,7),16);return `rgba(${e}, ${t}, ${r}, ${n})`}

exports.Portal = p;
exports.getStringMatch = d;
exports.hexToRGBA = b;
exports.useKeyboard = m;
